### PR TITLE
Torcharrow based training using RoBERTa model and SST2 classification dataset

### PR DESCRIPTION
**Pull request by parmeet**
Example end-2-end training for RoBERTa model with TorchArrow based text pre-processing.

### CODE REVIEW

File: examples/torcharrow/roberta_sst2_training_with_torcharrow.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    bpe = _ta.GPT2BPEEncoder(bpe_encoder, bpe_merge_ranks, _seperator, T.bytes_to_unicode(), True)
+    return bpe
+
+
+def init_ta_gpt2bpe_vocab():
+    vocab_path = "https://download.pytorch.org/models/text/roberta.vocab.pt"
+    vocab_path = get_asset_local_path(vocab_path)
+    vocab = torch.load(vocab_path)
+    ta_vocab = _ta.Vocab(vocab.get_itos(), vocab.get_default_index())
+    return ta_vocab
```
Comment (Nayef211) [inline examples/torcharrow/roberta_sst2_training_with_torcharrow.py:49]: Do we want to reuse the implementations from here?https://github.com/pytorch/text/blob/5f48259762aeb739e7d754dc12b868cdc66cc7ad/examples/data_pipeline/roberta_dataframe.py#L18-L45

File: examples/torcharrow/roberta_sst2_training_with_torcharrow.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        # Add EOS token to the end of sentence
+        self.add_eos = T.AddToken(token=2, begin=False)
+
+    def forward(self, input: ta.DataFrame) -> ta.DataFrame:
+        input["tokens"] = ta_F.bpe_tokenize(self.tokenizer, input["text"])
+        input["tokens"] = input["tokens"].list.slice(stop=254)
+        input["tokens"] = ta_F.lookup_indices(self.vocab, input["tokens"])
+        input["tokens"] = input["tokens"].transform(self.add_bos, format="python")
+        input["tokens"] = input["tokens"].transform(self.add_eos, format="python")
+        return input
```
Comment (Nayef211) [inline examples/torcharrow/roberta_sst2_training_with_torcharrow.py:73]: Should we reuse this class
https://github.com/pytorch/text/blob/5f48259762aeb739e7d754dc12b868cdc66cc7ad/examples/data_pipeline/roberta_dataframe.py#L48-L69

File: examples/torcharrow/roberta_sst2_training_with_torcharrow.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+learning_rate = 1e-5
+optim = AdamW(model.parameters(), lr=learning_rate)
+criteria = nn.CrossEntropyLoss()
+
+
+def train_step(input, target):
+    output = model(input)
+    loss = criteria(output, target)
+    print(float(loss))
```
Comment (Nayef211) [inline examples/torcharrow/roberta_sst2_training_with_torcharrow.py:113]: nit: do we want to keep the print here?

File: examples/torcharrow/roberta_sst2_training_with_torcharrow.py (hunk 1 above)
Comment (parmeet) [reply examples/torcharrow/roberta_sst2_training_with_torcharrow.py:49]: hmm, I was thinking of avoiding it for end-2-end training example to give better visibility into all the functionalities through one standalone example. Just to be clear, I didn't mean that we cannot break down standalone example into multiple modules, just that the data_pipeline is a standalone example in itself and is not explicitly intended to be part of end-2-end training script..Let me know if this make sense?

File: examples/torcharrow/roberta_sst2_training_with_torcharrow.py (hunk 2 above)
Comment (parmeet) [reply examples/torcharrow/roberta_sst2_training_with_torcharrow.py:73]: same comment as above..

File: examples/torcharrow/roberta_sst2_training_with_torcharrow.py (hunk 3 above)
Comment (parmeet) [reply examples/torcharrow/roberta_sst2_training_with_torcharrow.py:113]: not really :)

File: examples/torcharrow/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+make sure to use flag `USE_TORCH=1` during TorchArrow installation (this is also the reason why we cannot depend on
+nightly releases)
+
+```
+USE_TORCH=1 python setup.py install
+```
+
+#### TorchData
+
+To install TorchData follow instructions athttps://github.com/pytorch/data#installation
```
Comment (Nayef211) [inline examples/torcharrow/README.md:25]: nit: add a space after "at"

**Review by Nayef211 — APPROVED** [review]
LGTM

File: examples/torcharrow/roberta_sst2_training_with_torcharrow.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        # vocabulary converting tokens to IDs
+        self.vocab = init_ta_gpt2bpe_vocab()
+
+        # Add BOS token to the beginning of sentence
+        self.add_bos = T.AddToken(token=0, begin=True)
+
+        # Add EOS token to the end of sentence
+        self.add_eos = T.AddToken(token=2, begin=False)
+
+    def forward(self, input: ta.DataFrame) -> ta.DataFrame:
```
Comment (wenleix) [inline examples/torcharrow/roberta_sst2_training_with_torcharrow.py:67]: We can probably just to have a standalone `preproc` function, similar to https://github.com/pytorch/torchrec/blob/20f543ee3700f2ebb27be6e80d636bd5dd0d7f3c/examples/torcharrow/dataloader.py#L59-L94

Comment (parmeet) [reply examples/torcharrow/roberta_sst2_training_with_torcharrow.py:67]: SGTM, thanks for the suggestion @wenleix. Let me do a follow-up PR to do that.

**Pull request merged**